Automattic/woocommerce-payments
# Add Capture Charge button on payment details page

Fixes #4506 

#### Changes proposed in this Pull Request
* Add a button on Payment Details page to capture an authorized payment.

A glimpse at the capture flow: 

https://user-images.githubusercontent.com/1553182/198542855-c10f26e0-2a79-43b9-b4cb-814b0bf0fb5d.mov










#### Testing instructions
* Set displayCaptureAuthorizationSection to true [here](https://github.com/Automattic/woocommerce-payments/pull/5006/files#diff-a41f2e2fca1befa933534f9c72b4d1552d19af998a9dc2252f592d31e10b7a59R34)
* Go to WP Admin → Payments → Settings and make sure the checkbox "Issue an authorization on checkout, and capture later" is checked.
* Create a new order using WC frontend. (eg. buy a Beanie using a [test card](https://woocommerce.com/document/payments/testing/#test-card-numbers))
* Open the order details page from WP Admin → WooCommerce → Orders.
* Click the payment intent link to visit the order's payment details page.
* Verify that you see a notice "You need to capture this charge before" along with a capture button.
* Click the Capture button.
* Verify that the payment is successfully captured. (the notice disappears, a success message appears, timeline is refreshed)
* Refresh the page and verify that the capture notice doesn't appear.


-------------------


**Post merge**
